inputfiles/addedTypes.jsonc

```diff
@@ -1374,6 +1463,21 @@
             {
                 "name": "EventListenerOrEventListenerObject",
                 "overrideType": "EventListener | EventListenerObject"
+            },
+            {
+                "name": "AutoFillSection",
+                "overrideType": "`section-${string}`"
+            },
+            {
+                "name": "AutoFillField",
+                "overrideType": "AutoFillNormalField | AutoFillContactField | `${AutoFillContactKind} ${AutoFillContactField}`"
+            },
+            {
+                // See the full list of possible autofill values for the `autocomplete` attribute:
+                // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#determine-a-field's-category
+                // Full spec at https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill.
+                "name": "AutoFill",
+                "overrideType": "AutoFillBase | AutoFillField | `${AutoFillSection} ${AutoFillField}` | `${AutoFillAddressKind} ${AutoFillField}` | `${AutoFillSection} ${AutoFillAddressKind} ${AutoFillField}` | `${AutoFillSection} ${AutoFillAddressKind} ${AutoFillField} ${AutoFillCredentialField}` | `${AutoFillAddressKind} ${AutoFillField} ${AutoFillCredentialField}` | `${AutoFillSection} ${AutoFillField} ${AutoFillCredentialField}` | `${AutoFillField} ${AutoFillCredentialField}`"
```

Comment:
Hmm, so per the spec it's something like:

`AutofillSection? AutofillAddressKind? (AutoFillNormalField | (AutoFillContactKind? AutoFillContactField)) AutoFillCredentialField?`

And this is the result of spreading it to every possible combination.

It's hard to read and maintain, I wonder this could be better as the type system is very versatile nowadays. 🤔

Reply:
How about this?

```ts
type OptionalPrefixToken<T extends string> = `${T} ` | ""
type OptionalPostfixToken<T extends string> = ` ${T}` | ""

// (AutoFillNormalField | (AutoFillContactKind? AutoFillContactField))
type AutoFillField = AutoFillNormalField | `${OptionalPrefixToken<AutoFillContactKind>}${AutoFillContactField}`;

// AutofillSection? AutofillAddressKind? AutoFillField AutoFillCredentialField?
type AutoFill = `${OptionalPrefixToken<AutoFillSection>}${OptionalPrefixToken<AutoFillAddressKind>}${AutoFillField}${OptionalPostfixToken<AutoFillCredentialField>}`
```

Reply:
That seems not problematic for me. Someone else from Microsoft may disagree, though.

Reply:
I think it's fine to show the union instead of the alias.

Reply:
Thanks, I'll try to commit this when I get some time (I've broken my arm ^^).

Reply:
Oh no, take care 🤕

Reply:
@saschanaz: done. :)

Thanks, I agree it's much more readable and maintainable like that.